### PR TITLE
nested logic is more robust in this case

### PR DESCRIPTION
**Pull request by LAShemilt**

### CODE REVIEW

File: rfi_file_monitor/utils/__init__.py [hunk 1]
```diff
@@ -45,16 +45,18 @@ def class_in_object_iterable(iterable: Iterable, klass) -> bool:
...
+    if not defaults:
+        if not input or not input.strip():
+            return ['*']
+        else:
+            # multiple patterns are supported, provided they are separated with commas
+            return list(map(lambda x: x.strip(), input.split(',')))
     else:
-    # multiple patterns are supported, provided they are separated with commas
-        return list(map(lambda x: x.strip(), input.split(',')))
+         if input:
```
Comment (tschoonj) [inline rfi_file_monitor/utils/__init__.py:56]:
```suggestion
         if input and input.strip():
```

File: rfi_file_monitor/utils/__init__.py [hunk 2]
```diff
@@ -45,16 +45,18 @@ def class_in_object_iterable(iterable: Iterable, klass) -> bool:
...
-def get_patterns_from_string(input: str,defaults =[]) -> List[str]:
-    if not input or not input.strip() and not defaults:
-        return ['*']
-    elif input and defaults:
-         return list(map(lambda x: x.strip(), input.split(','))) + defaults
-    elif not input and defaults:
-        return defaults
+def get_patterns_from_string(input: str, defaults: List =None) -> List[str]:
+    if not defaults:
+        if not input or not input.strip():
```
Comment (tschoonj) [inline rfi_file_monitor/utils/__init__.py:50]: I would flip the if and else block, and use `if input and input.strip():`

**Review by tschoonj — CHANGES_REQUESTED** [review]

File: rfi_file_monitor/utils/__init__.py (hunk 2 above)
Comment (LAShemilt) [reply rfi_file_monitor/utils/__init__.py]: done in d3d544b

File: rfi_file_monitor/utils/__init__.py (hunk 1 above)
Comment (LAShemilt) [reply rfi_file_monitor/utils/__init__.py]: done in 211e554

**Review by tschoonj — APPROVED** [review]

**Pull request merged**